agent/proxycfg/snapshot.go

```diff
@@ -905,9 +911,25 @@ DOMAIN_LOOP:
 	return services, upstreams, compiled, err
 }
 
-func (c *configSnapshotAPIGateway) toIngressTLS() (*structs.GatewayTLSConfig, error) {
-	// TODO (t-eckert) this is dependent on future SDS work.
-	return &structs.GatewayTLSConfig{}, nil
+func (c *configSnapshotAPIGateway) toIngressTLS(key IngressListenerKey, listener structs.APIGatewayListener, bound structs.BoundAPIGatewayListener) (*structs.GatewayTLSConfig, error) {
+	if len(listener.TLS.Certificates) == 0 {
+		return nil, nil
+	}
+
+	for _, certRef := range bound.Certificates {
+		cert, ok := c.Certificates.Get(certRef)
+		if !ok {
+			continue
+		}
+		c.ListenerCertificates[key] = append(c.ListenerCertificates[key], *cert)
+	}
+
+	return &structs.GatewayTLSConfig{
+		Enabled:       true,
```

Comment:
Is there an edge case where the `listener.TLS.Certificates` is not empty but the `bound.Certificates` does not have a cert ref that matches in `c.Certificates`? Could we be enabling TLS without a legit cert?

Reply:
I think that should be alright. If we wind up enabling TLS and have no certs that match in our snapshot, what should wind up happening is that we'll see no certificates in the listener generation code but will still see the `enabled` flag and accordingly fallback to the basic ingress behavior which is "use the leaf cert".